src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -1,18 +1,41 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
 package frc.robot.subsystems;
-import edu.wpi.first.wpilibj.command.Subsystem;
+
 import edu.wpi.first.wpilibj.drive.MecanumDrive;
 import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
-public class DriveSubsystem extends Subsystem{
-    private WPI_TalonSRX m_frontLeft  = new WPI_TalonSRX(4);
-    private WPI_TalonSRX m_rearLeft   = new WPI_TalonSRX(7);
-    private WPI_TalonSRX m_frontRight = new WPI_TalonSRX(6);	
-    private WPI_TalonSRX m_rearRight  = new WPI_TalonSRX(8);
-    private MecanumDrive m_robotDrive = new MecanumDrive(m_frontLeft,m_rearLeft,m_frontRight,m_rearRight);
-    @Override
-    protected void initDefaultCommand() {
-        m_robotDrive.driveCartesian(0.0, 0.0, 0.0);
-    }
-    public void drive(double x, double y, double theta){
-        m_robotDrive.driveCartesian(x, y, theta);
-    }
+
+/**
+ * Add your docs here.
+ */
+public class DriveSubsystem extends BaseSubsystem{
+  // Put methods for controlling this subsystem
+  // here. Call these from Commands.
+
+  private WPI_TalonSRX m_frontLeft  = new WPI_TalonSRX(4);
+  private WPI_TalonSRX m_rearLeft   = new WPI_TalonSRX(7);
+  private WPI_TalonSRX m_frontRight = new WPI_TalonSRX(6);	
+  private WPI_TalonSRX m_rearRight  = new WPI_TalonSRX(8);
+  private MecanumDrive m_robotDrive = new MecanumDrive(m_frontLeft,m_rearLeft,m_frontRight,m_rearRight);
+    
+  @Override
+  public void initialize() {
+
```

Comment:
I wonder if we need to move the assignment of the motors here.  So we have:
m_frontLeft = new WPI_TalonSRX(4);
m_rearLeft   = new WPI_TalonSRX(7);
m_frontRight = new WPI_TalonSRX(6);
m_rearRight  = new WPI_TalonSRX(8);
m_robotDrive = new MecanumDrive(m_frontLeft,m_rearLeft,m_frontRight,m_rearRight);

Then the code in lines 20-24, we truncate delete everything from the equals sign on.  So a line looks like this:
private WPI_TalonSRX m_frontLeft;

etc.

Reply:
Additional note: this would apply to the other sub-systems below too.